presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -392,4 +392,43 @@ private static void checkRadix(long radix)
         checkCondition(radix >= MIN_RADIX && radix <= MAX_RADIX,
                 INVALID_FUNCTION_ARGUMENT, "Radix must be between %d and %d", MIN_RADIX, MAX_RADIX);
     }
+
+    @Description("The bucket number of a value given a lower and upper bound and the number of buckets")
+    @ScalarFunction("width_bucket")
+    @SqlType(StandardTypes.BIGINT)
+    public static long widthBucket(@SqlType(StandardTypes.DOUBLE) double operand, @SqlType(StandardTypes.DOUBLE) double bound1, @SqlType(StandardTypes.DOUBLE) double bound2, @SqlType(StandardTypes.BIGINT) long bucketCount)
+    {
+        checkCondition(bucketCount > 0, INVALID_FUNCTION_ARGUMENT, "bucketCount must be greater than 0");
+        checkCondition(!isNaN(operand), INVALID_FUNCTION_ARGUMENT, "operand must not be NaN");
+        checkCondition(isFinite(bound1), INVALID_FUNCTION_ARGUMENT, "first bound must be finite");
+        checkCondition(isFinite(bound2), INVALID_FUNCTION_ARGUMENT, "second bound must be finite");
+        checkCondition(bound1 != bound2, INVALID_FUNCTION_ARGUMENT, "bounds cannot equal each other");
+
+        long result = 0;
+
+        if (bound1 < bound2) {
+            if (operand < bound1) {
+                result = 0;
+            }
+            else if (operand >= bound2) {
+                result = Math.addExact(bucketCount, 1);
+            }
+            else {
+                result = (long) ((double) bucketCount * (operand - bound1) / (bound2 - bound1) + 1);
+            }
+        }
+        else if (bound1 > bound2) {
```

Comment:
This branch is equivalent to the one above with the bounds flipped, no? If so, I'd write the code as:

```java
long lower = Math.min(bound1, bound2);
long upper = Math.max(bound1, bound2);

if (operand < lower) {
    result = 0;
}
else if (operand >= upper) {
    result = Math.addExact(bucketCount, 1);
}
else {
    result = (long) ((double) bucketCount * (operand - lower) / (upper - lower) + 1);
}
```